resources/views/hardware/checkout.blade.php

```diff
@@ -140,11 +140,19 @@
                             <label for="note" class="col-md-3 control-label">
                                 {{ trans('general.notes') }}
                             </label>
+                            @if($snipeSettings->require_checkinout_notes=="1")
+                            <div class="col-md-8">
+                                <textarea class="col-md-6 form-control" id="note" required="true"
+                                          name="note">{{ old('note', $asset->note) }}</textarea>
+                                {!! $errors->first('note', '<span class="alert-msg" aria-hidden="true"><i class="fas fa-times" aria-hidden="true"></i> :message</span>') !!}
+                            </div>
+                            @else
```

Comment:
Since the only difference is adding `required=true` you should be able to re-use the existing block and throw the logic in a `@required`. Something like `@required($snipeSettings->require_checkinout_notes=="1")`.

[Docs](https://laravel.com/docs/10.x/blade#additional-attributes)

Reply:
We'll also want to surface the `required` value the way we do on other required elements so we get the yellow bar that indicates requiredness.

Reply:
Additionally, I don't see any checks on the backend or in the form request that checks for requiredness, which we'd probably want.

Reply:
so, the `required=true` bit is because we'd still want the note box, even if it _isn't_ required right? so, the box needs to populate either way.

Reply:
Yes, the note box should always be there, we just need to show the required indicator if it is actually required via the settings, and we should be checking for that requiredness on the backend. Might be easiest to add it to the form requests.

Reply:
right, i think that was in response to Marcus.
and yea, I'm doing the backend checking now. I'll peek the form requests.

Reply:
Another way of saying my original comment is that the original textarea can be re-used with `required="true"` conditionally added to it. I mentioned the helper method `@required` which I think is cleaner than `{{ $snipeSettings->require_checkinout_notes=="1" ? 'required="true" : '' }}`.

Re-using the existing block (textarea) will reduce the duplication that comes from copy/pasting the multi-line block and to only add the required stuff.

Snipe is saying we also need to conditionally add the orange warning for a required field like we have for a couple other elements on the page:
![CleanShot 2024-08-20 at 12 03 37@2x](https://github.com/user-attachments/assets/0334021d-8d8d-4ef8-8490-c291353787c2)

Reply:
ya, the orange bar is already done.

Reply:
fix in app.less